### ISSUE TITLE
Scanning project containing local packages 

### ISSUE THREAD

**Issue by PiotrPawlus** [peripheryapp/periphery#787]
### Description

I'm currently working on a project that contains one target but multiple schemes that represent my Local packages. CLI generates the YML file detecting all targets and schemes.  

### Steps to Reproduce

```yml
index_store_path:
- build/DerivedData/My_Project/Index.noindex/DataStore
retain_public: true
retain_objc_accessible: true
retain_objc_annotated: true
retain_assign_only_properties: true
retain_assign_only_property_types: true
retain_swift_ui_previews: true                        
retain_codable_properties: true
verbose: false
schemes:
- Ads
- MyFoundation
- MyProject
- Design
# And many many more 
skip_build: true
targets:
- MyProject
workspace: MyProject.xcworkspace
```

### Actual Behavior
The "main" project is scanned correctly. The local packages are not scanned.

### Environment
- periphery version `2.21.0`
- Xcode 15.4

### Additionally 
- I tried to build the project with `xcodebuild` for each of the targets but periphery won't allow me to provide dd for each of it for each of the schemes we have

**Comment by Xopoko** [comment]
I’m experiencing the exact same issue! 

Initially, I thought it was related to my settings, so I tried modifying various project build configurations (using Tuist). I even looked into the 'DataStore/V5' folder to see what was inside and found that it contains files from local packages. However, Periphery still only scans the files that are directly part of the project, ignoring those in the packages.

**Comment by ileitch** [comment]
As noted by @Xopoko, this should be resolved in the 3.0 beta.

**Comment by ileitch** [comment]
This issue should now be resolved in 3.0.0. Let me know if you continue to experience this issue and I will reinvestigate.
